assets/chezmoi.io/docs/reference/templates/directives.md

```diff
@@ -37,6 +37,22 @@ inherited by templates called from the file.
     # [[ "true" ]]
     ```
 
+## Format indent
+
+By default, chezmoi's `toJson`, `toToml`, and `toYaml` template functions use
+the default indent of two spaces. The indent can be overidden with:
+
+    chezmoi:template:format-indent=$VALUE
+
+`$VALUE` can be an integer number of spaces or a string.
```

Comment:
I wonder if this might not be better as two directives, `format-indent-width` and `format-indent`. They would be mutually exclusive. `format-indent-width` would be the integer number of spaces to use per level; `format-indent` would a a string, but would have to match `^[ \t]+$` to be valid. That is, `format-indent=""` would not be valid, nor would `format-indent=".."`.

(One could argue that it should be `(^ +$|^\t+$)` to prevent mixing tabs and spaces.)

Reply:
Thanks, done. I've not been super-strict with the parsing, but any errors should be quickly visible to the user.

Reply:
If we want to be strict about only allowing tabs or spaces, would something similar to how it's done in [EditorConfig](https://spec.editorconfig.org/#indentation-non-normative) be better? i.e.

```text
chezmoi:template:indent-style:spaces
chezmoi:template:indent-style:tabs
```

```text
chezmoi:template:indent-size:4

# and possibly also
chezmoi:template:tab-width:4
```

Reply:
I don't think we need to be super-strict on this, and that it should be sufficient to need only one directive to specify the indent. So, I'll leave this as-is.